lib/pure/options.nim

```diff
@@ -176,21 +310,239 @@ proc filter*[T](self: Option[T], callback: proc (input: T): bool): Option[T] =
   else:
     self
 
-proc `==`*(a, b: Option): bool =
-  ## Returns ``true`` if both ``Option``s are ``none``,
-  ## or if they have equal values
-  (a.isSome and b.isSome and a.val == b.val) or (not a.isSome and not b.isSome)
-
 proc `$`*[T](self: Option[T]): string =
   ## Get the string representation of this option. If the option has a value,
-  ## the result will be `Some(x)` where `x` is the string representation of the contained value.
-  ## If the option does not have a value, the result will be `None[T]` where `T` is the name of
-  ## the type contained in the option.
+  ## the result will be `Some(x)` where `x` is the string representation of the
+  ## contained value. If the option does not have a value, the result will be
+  ## `None[T]` where `T` is the name of the type contained in the option.
   if self.isSome:
     "Some(" & $self.val & ")"
```

Comment:
Will this work properly with strings? I seem to recall that `addQuoted` might need to be used in these cases.

Reply:
@Varriount that is a true criticism. But this is a problem unrelated to this PR.